File: d66-social/src/main/java/com/eroldmr/d66/constant/VoteType.java
package com.eroldmr.d66.constant;

import lombok.AllArgsConstructor;

/**
 * @author Mompati 'Patco' Keetile
 * @created 19-12-2022 @ 12:23
 */
@AllArgsConstructor
public enum VoteType {
  UP_VOTE(1),
  DOWN_VOTE(-1);
  private final Integer direction;
}

File: d66-social/src/main/kotlin/com/eroldme/d66/subredit/vote/constant/VoteType.kt
package com.eroldme.d66.subredit.vote.constant

/**
 * @author Mompati 'Patco' Keetile
 * @since 06-12-2022 @ 14:57
 *
 *
 */
enum class VoteType(private val direction: Int) {
    UP_VOTE(1),
    DOWN_VOTE(-1)
}